coqpp/coqpp_parse.mly

```diff
@@ -108,8 +108,13 @@ vernac_extend:
 ;
 
 tactic_extend:
-| TACTIC EXTEND IDENT tactic_level tactic_rules END
-  { TacticExt { tacext_name = $3; tacext_level = $4; tacext_rules = $5 } }
+| TACTIC EXTEND IDENT tactic_deprecated tactic_level tactic_rules END
+  { TacticExt { tacext_name = $3; tacext_deprecated = $4; tacext_level = $5; tacext_rules = $6 } }
+;
+
+tactic_deprecated:
+| { None }
+| DEPRECATED IDENT { Some $2 }
```

Comment:
This should probably be a string rather than an identifier.

Reply:
Why? You mean, it may need to be qualified?

Reply:
The use-case I have is as follows (excerpts):

```
let romega_depr = Vernacinterp.mk_deprecation ()

(* … *)

TACTIC EXTEND romega
DEPRECATED romega_depr
|  …
```

Using a string would require to put double quotes around the name of the constant `romega_depr`, right?

Reply:
Ah, I thought this was the message payload. So rather than STRING, you should then use the CODE token so that it can be an arbitrary piece of OCaml (wrapped between braces). The old macro system is quite limited, let's take advantage of the switch to be as expressive as possible.

Reply:
If I go for CODE, how do I print it?

Reply:
Just print the code field of the record.

Reply:
Indeed… Done. Thanks.